### PR TITLE
external: added missing realm, zonegroup and zone while interacting with users

### LINKED ISSUE #11806: External cluster: Create Multiple admin ops user in multisite environment
> And I am assuming default zone/zonegroup does not refers to master zone/zonegroup in the realm.

correct. multisite was designed for disaster recovery, where you'd generally only put one zone on each ceph cluster. so the primary zone would be the default on its cluster, and the secondary zone would be the default on the other cluster. in that scenario, the defaults make things simpler to manage. but whenever you have multiple zones or zonegroups on a cluster, defaults are likely to confuse things

> do you think is it better to include rgw-zone / rgw-zonegroup for this helper script in external ceph cluster usecases?

i think the script will work more reliably if you require the zone and zonegroup options

_Originally posted by @cbodley in https://github.com/rook/rook/pull/11584#discussion_r1121698957_

### PR DESCRIPTION
**Pull request by sergioperez**
**Description of your changes:**
Add missing usage of realm, zone and zonegroup while retrieving user info and adding user capabilities.

**Which issue is resolved by this Pull Request:**
Resolves #11806
Complement of #12037

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: deploy/examples/create-external-cluster-resources.py [hunk 1]
```diff
@@ -1152,6 +1152,12 @@ def create_rgw_admin_ops_user(self):
                     "info",
                     "--uid",
                     self.EXTERNAL_RGW_ADMIN_OPS_USER_NAME,
+                    "--rgw-realm",
+                    self._arg_parser.rgw_realm_name,
```
Comment (travisn) [inline deploy/examples/create-external-cluster-resources.py:1156]: If these parameters are not specified, will the command still succeed with defaults (as long as the defaults are valid)? Or if these args are now required, perhaps we would need a doc update.

**Review by travisn — CHANGES_REQUESTED** [review]
Please add the prefix to the commit title, and also add a description paragraph to the commit as described in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).

File: deploy/examples/create-external-cluster-resources.py (hunk 1 above)
Comment (parth-gr) [reply deploy/examples/create-external-cluster-resources.py:1156]: The command succeeds is what I have checked.
And it makes it in `default`  discussed here https://github.com/rook/rook/pull/12037#discussion_r1163333816.
@alimaredia to keep me honest.

File: deploy/examples/create-external-cluster-resources.py [hunk 2]
```diff
@@ -1152,6 +1152,12 @@ def create_rgw_admin_ops_user(self):
                     "info",
                     "--uid",
                     self.EXTERNAL_RGW_ADMIN_OPS_USER_NAME,
+                    "--rgw-realm",
```
Comment (parth-gr) [inline deploy/examples/create-external-cluster-resources.py:1155]: @thotz can you confirm this,
Do we need to pass the flag if a user is already created?

File: deploy/examples/create-external-cluster-resources.py [hunk 3]
```diff
@@ -1190,6 +1196,12 @@ def create_rgw_admin_ops_user(self):
             self.EXTERNAL_RGW_ADMIN_OPS_USER_NAME,
             "--caps",
             "info=read",
+            "--rgw-realm",
```
Comment (parth-gr) [inline deploy/examples/create-external-cluster-resources.py:1199]: We are just updating the caps,
Still not sure if we need these flags 
@thotz please confirm

**Comment by parth-gr** [comment]
@alimaredia @thotz can you confirm this?
> I see in the docs, we need to create --system user to facilitate authentication between daemons,
In this way i guess probably one system user is required,

https://docs.ceph.com/en/quincy/radosgw/multisite/#create-a-system-user
https://docs.ceph.com/en/quincy/radosgw/multisite/#migrating-a-single-site-deployment-to-multi-site

> For the external ceph cluster, this will already configured by the admin IMO, like script is not creating zone/zonegroup/realm etc. This is basic prerequisite for the multisite set up


https://github.com/rook/rook/pull/12037#issuecomment-1500391900


But the rgw-admin ops user is created by the script so do we need this??


Second Is there can a setup where I can test all these changes related to Multisite??

**Comment by parth-gr** [comment]
@thotz and the script can only use one --rgw-pool-prefix, for a ceph cluster, does that can be changed in any case of multisite,

If we try to change the client.healthchecker user tries to fail as it is already created with one rg pool,
So do we need to support it for multiple prefix or for one cluster only one is fine
https://github.com/rook/rook/blob/master/deploy/examples/create-external-cluster-resources.py#L1088

**Review by thotz — COMMENTED** [review]
The changes in the PR looks valid, sorry I have missed to inform this earlier

**Comment by thotz** [comment]
> @thotz and the script can only use one --rgw-pool-prefix, for a ceph cluster, does that can be changed in any case of multisite,
> 
> If we try to change the client.healthchecker user tries to fail as it is already created with one rg pool, So do we need to support it for multiple prefix or for one cluster only one is fine https://github.com/rook/rook/blob/master/deploy/examples/create-external-cluster-resources.py#L1088
As discussed in the chat this might be different case which need to discussed seperately

**Comment by thotz** [comment]
> @alimaredia @thotz can you confirm this?
> 
> > I see in the docs, we need to create --system user to facilitate authentication between daemons,
> > In this way i guess probably one system user is required,

Thats will already created in the external cluster, we don't need to worry about that
> 
> https://docs.ceph.com/en/quincy/radosgw/multisite/#create-a-system-user https://docs.ceph.com/en/quincy/radosgw/multisite/#migrating-a-single-site-deployment-to-multi-site
> 
> > For the external ceph cluster, this will already configured by the admin IMO, like script is not creating zone/zonegroup/realm etc. This is basic prerequisite for the multisite set up
> 
> [#12037 (comment)](https://github.com/rook/rook/pull/12037#issuecomment-1500391900)
> 
> But the rgw-admin ops user is created by the script so do we need this??

The admins ops is used by Rook internally, nothing related to multisite settings
> 
> Second Is there can a setup where I can test all these changes related to Multisite??

I recommend set up multisite cluster with Rook and import it in different namespace/minikube cluster as external

**Pull request closed without merging**